src/template.cc

```diff
@@ -226,7 +226,7 @@ char *find_and_replace_templates(const char *inbuf) {
       outlen += len;
       *o = '\0';
       outbuf = static_cast<char *>(realloc(outbuf, outlen * sizeof(char)));
-      strncat(outbuf, tmpl_out, len);
+      strcat(outbuf, tmpl_out);
```

Comment:
This was also a warning because the new compiler can tell that `len` is derived from `strlen(tmpl_out)` which is then the same as using `strcat` directly.